dbms/src/Interpreters/ActionsVisitor.cpp

```diff
@@ -195,12 +195,22 @@ SetPtr makeExplicitSet(
     return set;
 }
 
-static String getUniqueName(const Block & block, const String & prefix)
+static String getUniqueName(ActionsVisitor::Data & data, const String & prefix)
 {
-    int i = 1;
-    while (block.has(prefix + toString(i)))
-        ++i;
-    return prefix + toString(i);
+    auto & block = data.getSampleBlock();
+    auto result = prefix;
```

Comment:
It looks like we'll get the first one without suffix. Is it expected?

Reply:
Yep, I wanted to have a name without suffix if we only have one column with this prefix. Ideally, this function generates column ids and not display names, so this is not very important. But we don't have clear separation between these concepts yet, so I wrote it this way.